src/App.tsx

```diff
@@ -1,35 +1,276 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { Button } from "./components/button";
+import { Input } from "./components/input";
+import {
+  getLottoPrizeCountByPrize,
+  Lotto,
+  LOTTO_PRICE,
+  LottoPrize,
+  useLottoStore,
+  WinningLotto,
+} from "./features/lotto";
+import { useId } from "react";
+import { useInput } from "./hooks/use-input";
+import { z } from "zod";
+import { formatDateString, objectEntries } from "./lib/utils";
+import { getLottoHistory, LottoHistoryEntry } from "./features/history";
+import { getMostFrequentWinningLottoNumber } from "./features/statistics";
 
-function App() {
-  const [count, setCount] = useState(0)
+export const App = () => {
+  const amountInputId = useId();
+  const amountInput = useInput();
+
+  const amountInputSchema = z.string().refine((value) => {
+    return parseInt(value) % LOTTO_PRICE === 0;
+  });
+
+  const {
+    purchaseLottoByAmount,
+    showResult,
+    startOver,
+    purchasedLotto,
+    winningLotto,
+  } = useLottoStore();
+
+  const onSubmit: React.FormEventHandler<HTMLFormElement> = (event) => {
+    event.preventDefault();
+
+    try {
+      amountInputSchema.parse(amountInput.value);
+
+      const lottoAmount = Math.floor(parseInt(amountInput.value) / LOTTO_PRICE);
+
+      purchaseLottoByAmount(lottoAmount);
```

Comment:
p3: 함수 시그니처가 요렇게 되면 어떨까 싶기도 하네요~
```suggestion
      purchaseLotto(lottoAmount);
```